### ISSUE TITLE
Shizuku support possible?

### ISSUE THREAD

**Issue by Steve-Brule** [Arian04/android-hid-client#17]
Hi! I always dreamed of an app like this, but sadly can't root my phone.

Is this app able to function with ADB permissions? If so, it'd be amazing to see shizuku support implemented.

**Comment by Arian04** [comment]
hmm. I have some ideas on how i might be able to get it working but it depends on how much Shizuku lets me do. I could *maybe* see if I could add HID functions to the default android gadget but I haven't checked if that would work without root or even at all

**Comment by Arian04** [comment]
thanks so much for this! I'm not too sure if it's possible to add Shizuku support (still looking into it), but either way, I just got my HID functions working with the standard gadget, which simplifies so much!! Now, the standard Android gadget shouldn't have to be disabled at all once I fully implement this. It simplifies SOOO much.

**Comment by Steve-Brule** [comment]
Haven't really done anything, but you're welcome regardless :)

**Comment by Arian04** [comment]
So after the simplification, my app only uses root in these places
- Adding HID functions to the default Android USB gadget (I think I only need `system` permissions for this though)
- Setting `/dev/hidg0`/`/dev/hidg1` file mode permissions, SELinux context, and adding an SELinux policy (I don't think I can do this without root, BUT maybe I can find a way for it to not be necessary)
- Grabbing debug logs (unnecessary, wasn't considering non-root users when I wrote it)

So basically, I might be able to get it working if I can get `system` permissions. I'll try to look into this further sometime soon.

**Comment by Arian04** [comment]
as far as I can tell, this is not possible :(

if anyone thinks I'm wrong about that or something changes that makes it so this could be possible in the future, feel free to reply under this issue and i'll consider re-opening it. Like I said, I don't use root in very many spots for this app, so if this becomes possible to do without root, code-wise, it should be an incredibly simple change.
